### PR TITLE
Change behavior of SocketsHttpHandler for redirects

### LINKED ISSUE #27285: SocketsHttpHandler: Correct behavior when MaxAutomaticRedirections is exceeded?
Currently, if MaxAutomaticRedirections is exceeded, we throw an exception.  This matches WinHttpHandler/CurlHandler.

This does not match full framework behavior, though.  Framework will simply stop redirecting and return the last 3xx response.

The latter seems slightly better to me.  However, it also means the user needs to handle this case and not assume a 200 response.  They pretty much have to do this anyway since the response could always be 4xx or 5xx anyway, but it's still one more situation for them to handle.

See HttpClientHandlerTest.GetAsync_MaxAutomaticRedirectionsNServerHops_ThrowsIfTooMany.

### PR DESCRIPTION
**Pull request by davidsh**
When exceeding the max number of redirecs, .NET Framework will
not throw an exception. Instead, it will return the 3xx response.
This PR changes the behavior of SocketsHttpHandler to match.

Fixes #27285

### CODE REVIEW

**Comment by davidsh** [comment]
cc: @dotnet/ncl

**Comment by davidsh** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build

File: src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs [hunk 1]
```diff
@@ -33,14 +33,22 @@ public RedirectHandler(int maxAutomaticRedirections, HttpMessageHandler initialI
             Uri redirectUri;
             while ((redirectUri = GetUriForRedirect(request.RequestUri, response)) != null)
             {
-                response.Dispose();
-
                 redirectCount++;
+
                 if (redirectCount > _maxAutomaticRedirections)
                 {
-                    throw new HttpRequestException(SR.net_http_max_redirects);
```
Comment (pjanotti) [inline src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs]: Delete the string from resx or there is some code still using it?

Comment (davidsh) [reply src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/RedirectHandler.cs]: I'll remove the string. There are no other uses of it.

**Comment by davidsh** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build
@dotnet-bot test Outerloop Linux x64 Debug Build

**Comment by davidsh** [comment]
CI failures unrelated to PR.

**Pull request merged**